primitives/teerex/src/lib.rs

```diff
@@ -35,31 +35,110 @@ impl Default for SgxBuildMode {
 }
 
 #[derive(Encode, Decode, Copy, Clone, PartialEq, Eq, sp_core::RuntimeDebug, TypeInfo)]
-pub enum AttestationMethod {
-	Dcap,
+pub enum SgxAttestationMethod {
+	Skip(bool),
 	Ias,
-	Skip,
+	Dcap(bool),
+}
+
+impl Default for SgxAttestationMethod {
+	fn default() -> Self {
+		SgxAttestationMethod::Skip(false)
+	}
+}
```

Comment:
I am not sure if you can derive the default as shown below if the enum variant contains a field.

Reply:
seems to be possible